src/adhocracy_frontend/adhocracy_frontend/static/js/Packages/ResourceActions/ResourceActions.ts

```diff
@@ -0,0 +1,145 @@
+import * as AdhConfig from "../Config/Config";
+import * as AdhMovingColumns from "../MovingColumns/MovingColumns";
+import * as AdhPermissions from "../Permissions/Permissions";
+import * as AdhTopLevelState from "../TopLevelState/TopLevelState";
+import * as AdhUtil from "../Util/Util";
+
+var pkgLocation = "/ResourceActions";
+
+export var resourceActionsDirective = (
+    adhPermissions : AdhPermissions.Service,
+    adhConfig: AdhConfig.IService
+) => {
+    return {
+        restrict: "E",
+        scope: {
+            resourcePath: "@",
+            share: "=?",
+            delete: "=?",
+            print: "=?",
+            report: "=?",
+            cancel: "=?",
+            edit: "=?"
+        },
+        templateUrl: adhConfig.pkg_path + pkgLocation + "/ResourceActions.html",
+        link: (scope) => {
+            adhPermissions.bindScope(scope, () => scope.resourcePath && AdhUtil.parentPath(scope.resourcePath), "proposalItemOptions");
+        }
+    };
+};
+
+export var reportActionDirective = () => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"report();\">{{ \"TR__REPORT\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            class: "@"
+        },
+        link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
+            scope.report = () => {
+                column.toggleOverlay("abuse");
+            };
+        }
+    };
+};
+
+export var shareActionDirective = () => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"report();\">{{ \"TR__SHARE\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            class: "@"
+        },
+        link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
+            scope.report = () => {
+                column.toggleOverlay("share");
+            };
+        }
+    };
+};
+
+export var deleteActionDirective = () => {
+    return {
+        restrict: "E",
+        template: "<a class=\"{{class}}\" href=\"\" data-ng-click=\"delete();\">{{ \"TR__DELETE\" | translate }}</a>",
+        require: "^adhMovingColumn",
+        scope: {
+            resourcePath: "@",
+            class: "@"
+        },
+        link: (scope, element, attrs, column : AdhMovingColumns.MovingColumnController) => {
+            scope.delete = () => {
+                column.$broadcast("triggerDelete", scope.proposalUrl);
```

Comment:
Would be a shame if we didn't standardise stuff like this, otherwise we have to just have a pass link function